apex/contrib/csrc/nccl_allocator/NCCLAllocator.cpp

```diff
@@ -0,0 +1,437 @@
+#include <c10/cuda/CUDACachingAllocator.h>
+#include <c10/cuda/CUDAGuard.h>
+#include <c10/util/ApproximateClock.h>
+#include <iostream>
+#include <mutex>
+#include <nccl.h>
+
+#include "NCCLAllocator.h"
+
+#if defined(NCCL_MAJOR) && \
+    ((NCCL_MAJOR > 2) || (NCCL_MAJOR == 2) && (NCCL_MINOR >= 19))
+#define NCCL_SUPPORTS_UB
+#endif
+
+#define C10_NCCL_CHECK(cmd)                                               \
```

Comment:
couldn't we include this one by including some c10 header?

Reply:
I guess we could could include `torch/csrc/distributed/c10d/NCCLUtils.hpp`, but I did not want to do, as that file may cause issues.